jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java

```diff
@@ -77,7 +96,9 @@ String getPassword() {
 
     @Nullable @Parameter private String credHelper;
 
-    @Parameter private AuthConfiguration auth = new AuthConfiguration();
+    @Parameter
+    private AuthConfiguration auth =
+        new AuthConfiguration("<from><auth><username>", "<from><auth><password>");
```

Comment:
This can be done in the same manner:

`new AuthConfiguration(descriptor + "<auth>");`
```
private AuthConfiguration(String descriptor) {
      this.usernameDescriptor = descriptor + "<username>";
      this.passwordDescriptor = descriptor + "<password>";
```

And below for example:

`new FromConfiguration("<from>")`

This way, the same hierarchical model can be applied to the other parameters and fetched via a "getDescriptor" method in other logging code.

Reply:
I'll change it for the auth related stuff, but I don't want to get too out of scope with this PR by changing up the other parameters.

Reply:
Sounds good.